server/src/main/java/com/cloud/api/query/QueryManagerImpl.java

```diff
@@ -646,7 +646,16 @@
 
         Ternary<Long, Boolean, ListProjectResourcesCriteria> domainIdRecursiveListProject = new Ternary<Long, Boolean, ListProjectResourcesCriteria>(cmd.getDomainId(), cmd.isRecursive(), null);
 
-        _accountMgr.buildACLSearchParameters(caller, null, cmd.getAccountName(), cmd.getProjectId(), permittedAccounts, domainIdRecursiveListProject, listAll, false);
+        Long projectId = cmd.getProjectId();
+        if (resourceType.equalsIgnoreCase("project") && projectId == null) {
+            try {
+                projectId = Long.parseLong(resourceId);
+            } catch (NumberFormatException e) {
+                projectId = _projectDao.findByUuid(resourceId).getId();
```

Comment:
What if the uuid is invalid ?

Reply:
it throws an exception: garbage in, garbage out. should we handle that? I think it is the exception @rhtyd is pointing at. How do you suggest this should be handled?